workspaces/core/src/combobox/Combobox.tsx

```diff
@@ -1,180 +1,544 @@
   size = 'md',
-  className = '',
+  className,
+  onClear,
+  multiple,
+  nullable,
   position = 'bottom-start',
-  label,
   placeholder,
   displayValue,
-}) => {
-  const multiple = false;
-  const [anchorEl, setAnchorEl] = useState<Element | null>(null);
-  const [popperEl, setPopperEl] = useState<HTMLElement | null>(null);
+  defaultValue,
+  ref,
+  ...rest
+}: ComboboxRootProps) => {
+  const [anchorEl, setAnchorEl] = React.useState<Element | null>();
+  const [popperEl, setPopperEl] = React.useState<HTMLElement | null>();
+  const [isInputFocused, setIsInputFocused] = React.useState<boolean>(false);
 
-  const { styles, attributes } = usePopper(anchorEl, popperEl, {
+  let { styles, attributes } = usePopper(anchorEl, popperEl, {
     placement: position,
   });
 
+  const states = {
+    value: value,
+    displayValue: displayValue,
+    isError: isError,
+    size: size,
+    disabled: disabled,
+    input: {
+      isFocused: isInputFocused,
+      setIsFocused: setIsInputFocused,
+    },
+    multiple: multiple,
+    onClear: onClear,
+    popper: {
+      styles: styles,
+      attributes: attributes,
+      setAnchor: setAnchorEl,
+      setPopper: setPopperEl,
+    },
+  };
+
+  const childrens =
+    typeof children !== 'function' ? React.Children.toArray(children) : [];
+  const callableChildren = typeof children === 'function' && children;
+
+  return (
+    <ComboboxContext.Provider value={states}>
+      <div
+        className={mergeClassnames('w-full relative', className)}
+      >
+        <ComboboxHeadlessUI
+          value={value as {}[]}
+          onChange={onChange}
+          disabled={disabled}
+          multiple={multiple as true}
+          nullable={nullable as true}
+          ref={ref as React.Ref<HTMLElement>}
+          {...rest}
+        >
+          {({ open }) => (
+            <>
+              {typeof children === 'function'
+                ? callableChildren && callableChildren({ open })
+                : childrens?.map((ch) => ch)
+              }
+            </>
+          )}
+        </ComboboxHeadlessUI>
+      </div>
+    </ComboboxContext.Provider>
+  );
+};
+
+const Trigger = ({
+  children,
+  className,
+  innerLabel,
+}: WithChildren<SelectProps>) => {
+  const { size, input, popper, disabled, isError } = useComboboxContext('Combobox.Trigger');
+
   return (
-    <div className={mergeClassnames('w-full relative', className)}>
-      <HeadlessCombobox
-        value={value}
-        onChange={onChange}
+    <div
+      tabIndex={-1}
+      className={mergeClassnames(
+        'relative',
+        'flex flex-nowrap w-full align-middle items-center rounded-lg py-2 px-3 bg-gohan gap-x-2',
+        size === 'sm' && 'py-1.5 px-2 rounded',
+        (size === 'sm' || (innerLabel !== undefined && innerLabel)) && 'py-2 px-3 rounded-md gap-x-3',
+        size === 'lg' && 'py-3 px-3',
+        (size === 'xl' || (innerLabel !== undefined && innerLabel)) && 'py-3 px-4 rounded-xl gap-x-4',
```

Comment:
I think it makes sense to put this in a separate helper `getSize(),` to make code more clearly readable